Utils/Functions/Net/Download/script.js

```diff
@@ -1,164 +1,184 @@
-Downloader.prototype.get = function () {
-    if (!this._message) {
-        this._message = tr("Please wait while {0} is downloaded...", this._fetchFileNameFromUrl(this._url));
+    /**
+     * Sets the checksum
+     *
+     * @param {string} checksum The checksum which shall be used to verify the download
+     * @returns {Downloader} The Downloader object
+     */
+    checksum(checksum) {
+        this._checksum = checksum;
+        return this;
     }
 
-    if (this._wizard) {
-        var progressBar = this._wizard.progressBar(this._message);
+    /**
+     * Sets the download message text
+     *
+     * @param {string} message The download message
+     * @returns {Downloader} The Downloader object
+     */
+    message(message) {
+        this._message = message;
+        return this;
     }
 
-    if (this._onlyIfUpdateAvailable) {
-        if (!this._downloader.isUpdateAvailable(this._localDestination, this._url)) {
-            print(this._localDestination + " already up-to-date.");
-            return;
-        }
+    /**
+     * Sets the http headers
+     *
+     * @param {{}} headers The http headers
+     * @returns {Downloader} The Downloader object
+     */
+    headers(headers) {
+        this._headers = headers;
+        return this;
     }
 
-    if (this._localDestination) {
-        var downloadFile = this._downloader.get(this._url, this._localDestination, this._headers, function (progressEntity) {
-            if (progressBar) {
-                progressBar.accept(progressEntity);
-            }
-        });
+    /**
+     * Sets the download destination
+     *
+     * @param {string} localDestination The destination of the download. If it is a directory, the file will be placed inside
+     * @returns {Downloader} The Downloader object
+     */
+    to(localDestination) {
+        this._localDestination = localDestination;
+        return this;
+    }
 
-        if (this._checksum) {
-            var fileChecksum = new Checksum()
-                .wizard(this._wizard)
-                .of(this._localDestination)
-                .method(this._algorithm)
-                .get();
+    /**
+     * Specifies if the download shall be executed only if a newer version is available
+     *
+     * @param {boolean} onlyIfUpdateAvailable true the download shall be executed only if a newer version is available
+     * @returns {Downloader} The Downloader object
+     */
+    onlyIfUpdateAvailable(onlyIfUpdateAvailable) {
+        this._onlyIfUpdateAvailable = onlyIfUpdateAvailable;
+        return this;
+    }
 
-            if (fileChecksum != this._checksum) {
-                var checksumErrorMessage = tr("Error while calculating checksum for \"{0}\". \n\nExpected = {1}\nActual = {2}",
-                    this._localDestination,
-                    this._checksum,
-                    fileChecksum);
+    /**
+     * Gets the content of the downloaded file
+     *
+     * @returns {String} The content of downloaded file
```

Comment:
@qparis @plata I'm not sure if it is a good idea to return file content here. I think this class is normally used to download binary files or other files that don't contain UTF-8 content.

For that reason I think it is better to return the local file path instead. In addition we can thing about renaming the method to `download`

Reply:
Makes sense but I'm not sure about the impact of this change.